pkg/console/controllers/router/controller.go

```diff
@@ -0,0 +1,133 @@
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	operatorclientv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	controllerWorkQueueKey = "router-ca-sync-work-queue-key"
+	controllerName         = "ConsoleRouterCASyncController"
+)
+
+type RouterCASyncController struct {
+	operatorConfigClient operatorclientv1.ConsoleInterface
+	configMapClient      coreclientv1.ConfigMapsGetter
+	// events
+	cachesToSync []cache.InformerSynced
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+func NewRouterCASyncController(
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	corev1Client coreclientv1.CoreV1Interface,
+	configMapInformer coreinformersv1.ConfigMapInformer,
+	// events
+	recorder events.Recorder,
+) *RouterCASyncController {
+	corev1Client.ConfigMaps(api.OpenShiftConsoleNamespace)
+
+	ctrl := &RouterCASyncController{
+		operatorConfigClient: operatorConfigClient,
+		configMapClient:      corev1Client,
+		// events
+		recorder:     recorder,
+		cachesToSync: nil,
+		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
+	}
+
+	configMapInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	ctrl.cachesToSync = append(ctrl.cachesToSync, configMapInformer.Informer().HasSynced)
+	return ctrl
+}
+
+func (c *RouterCASyncController) sync() error {
+	startTime := time.Now()
+	klog.V(4).Infof("started syncing route-ca configmap %q (%v)", api.RouterCAConfigMapName, startTime)
+	defer klog.V(4).Infof("finished syncing route-ca configmap %q (%v)", api.RouterCAConfigMapName, time.Since(startTime))
+
+	operatorConfig, err := c.operatorConfigClient.Get(api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfig.Spec.ManagementState {
+	case operatorsv1.Managed:
+		klog.V(4).Infoln("console is in a managed state: syncing router-ca configmap")
+	case operatorsv1.Unmanaged:
+		klog.V(4).Infoln("console is in an unmanaged state: skipping router-ca configmap sync")
+		return nil
+	case operatorsv1.Removed:
+		klog.V(4).Infoln("console is in a removed state: deleting router-ca configmap")
+		return c.removeRouterCAConfigMap()
+	default:
+		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
+	}
+
+	return err
+}
+
+func (c *RouterCASyncController) removeRouterCAConfigMap() error {
+	klog.V(2).Info("deleting router-ca configmap")
+	defer klog.V(2).Info("finished deleting router-ca configmap")
+	return c.configMapClient.ConfigMaps(api.OpenShiftConsoleNamespace).Delete(api.RouterCAConfigMapName, &metav1.DeleteOptions{})
+}
+
+func (c *RouterCASyncController) Run(workers int, stopCh <-chan struct{}) {
```

Comment:
Looks good. We can choose to use the `boilerplate` package to eliminate some of this repetitive stuff, but I have no qualms with this right now.